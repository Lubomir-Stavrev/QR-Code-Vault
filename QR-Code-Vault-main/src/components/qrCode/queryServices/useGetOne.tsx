import {useMutation} from 'react-query';
import storageServices from '../../../services/encryptedStorage';

export const useGetOne = () => {
  const getOne = (qrDataId: string) => {
    return storageServices.getOneQRCode(qrDataId);
  };
  const {mutate, isLoading, isError, isSuccess, data} = useMutation(getOne);

  return {mutate, isLoading, isError, isSuccess, data};
};
