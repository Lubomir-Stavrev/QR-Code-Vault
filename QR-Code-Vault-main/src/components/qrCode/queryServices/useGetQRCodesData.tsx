import {useQuery} from 'react-query';
import storageServices from '../../../services/encryptedStorage';

export const useGetQRCodesData = () => {
  const getQRCodes = () => {
    return storageServices.getQRCodes();
  };
  const {isLoading, isError, isSuccess, data} = useQuery('getQRCodesData', () =>
    getQRCodes(),
  );

  return {isLoading, isError, isSuccess, data};
};
