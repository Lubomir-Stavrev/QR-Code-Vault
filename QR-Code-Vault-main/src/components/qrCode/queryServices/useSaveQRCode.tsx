import {useMutation} from 'react-query';
import storageServices from '../../../services/encryptedStorage';

interface Props {
  navigate: (text: string) => void;
}

export const useSaveQRCode = (navigation: Props) => {
  const saveQRCodeRequest = (data: string) => {
    return storageServices.addQRCode(data);
  };
  const {mutate, isLoading, isError, isSuccess, data} = useMutation(
    saveQRCodeRequest,
    {
      onSuccess: () => {
        navigation.navigate('QRCodeMenu');
      },
    },
  );

  return {mutate, isLoading, isError, isSuccess, data};
};
