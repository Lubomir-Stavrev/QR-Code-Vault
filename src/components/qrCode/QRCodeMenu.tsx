import React, { useState, FC, useEffect } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
} from 'react-native';

import QRCodeScan from './QRCodeScan';
import UserQrCodes from './UserQrCodes';
import styles from '../../styles/QRCodeStyles'

const QRCodeRoute: FC = () => {
  const [optionsViewState, setOptionViewState] = useState<boolean | null>(true);
  const [isAddQRCode, setIsAddQRCode] = useState<boolean | null>(false);

  const handleQRCodeOptionPressed = (isAddQRCodeState: boolean) => {
    setIsAddQRCode(isAddQRCodeState);
    setOptionViewState(false);
  };

  const goToOptions = () => {

    setOptionViewState(true);
  };

  return (
    <View style={styles.qrContainer}>
      <View >
        {optionsViewState ? (
          <View
            style={styles.qrMenuContainer}>
            <Text
              style={styles.qrMenuTitle}>
              QR Code Vault
            </Text>
            <View >
              <TouchableOpacity onPress={() => handleQRCodeOptionPressed(true)}>
                <View style={styles.menuButtonContainer}>
                  <Text style={styles.menuButtonText}>Scan QR CODE</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() => handleQRCodeOptionPressed(false)}>
                <View style={styles.menuButtonContainer}>
                  <Text style={styles.menuButtonText}>View QR code collection</Text>
                </View>
              </TouchableOpacity>
            </View>
          </View>
        ) : isAddQRCode ? (
          <QRCodeScan goToOptions={goToOptions} />
        ) : (
          <UserQrCodes goToOptions={goToOptions} />
        )}
      </View>
    </View>
  );
};



export default QRCodeRoute;
