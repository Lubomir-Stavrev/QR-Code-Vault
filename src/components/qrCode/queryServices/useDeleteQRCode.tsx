import {useMutation, useQueryClient} from 'react-query';
import storageServices from '../../../services/encryptedStorage';

export const useDeleteQRCode = () => {
  const queryClient = useQueryClient();

  const deleteQrCodeFromStorage = (qrCode: string) => {
    return storageServices.deleteQRCode(qrCode);
  };
  const {mutate, isLoading, isError, isSuccess, data} = useMutation(
    deleteQrCodeFromStorage,
    {
      onSuccess: () => {
        queryClient.invalidateQueries('getQRCodesData');
      },
    },
  );

  return {mutate, isLoading, isError, isSuccess, data};
};
