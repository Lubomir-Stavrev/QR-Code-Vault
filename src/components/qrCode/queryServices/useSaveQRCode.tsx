import {useMutation} from 'react-query';
import storageServices from '../../../services/encryptedStorage';
import googleDriveServices from '../../../services/googleDriveServices';

interface Props {
  navigate: (text: string) => void;
}

export const useSaveQRCode = (navigation: Props) => {
  const saveQRCodeInDrive = async (data: string) => {
    const isSignedIn = await storageServices.isUserSignedInWithGoogle();
    if (isSignedIn) {
      return googleDriveServices.addQRCode(data);
    }
  };
  const saveQRCodeRequest = (data: string) => {
    saveQRCodeInDrive(data);
    return storageServices.addQRCode(data);
  };
  const {mutate, isLoading, isError, isSuccess, data} = useMutation(
    saveQRCodeRequest,
    {
      onSuccess: () => {
        navigation.navigate('QRCodeMenu');
      },
    },
  );

  return {mutate, isLoading, isError, isSuccess, data};
};
