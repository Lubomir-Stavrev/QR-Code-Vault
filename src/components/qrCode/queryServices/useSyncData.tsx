import {useMutation} from 'react-query';
import googleDriveServices from '../../../services/googleDriveServices';

export const useSyncData = () => {
  const syncDataWithDrive = async () => {
    return await googleDriveServices.syncDataWithDrive();
  };
  const {
    mutate: syncData,
    isLoading,
    isError,
    isSuccess,
    data,
  } = useMutation(syncDataWithDrive);

  return {syncData, isLoading, isError, isSuccess, data};
};
