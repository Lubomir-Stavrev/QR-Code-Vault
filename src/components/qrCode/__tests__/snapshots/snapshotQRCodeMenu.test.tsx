import React from 'react';
import QRCodeMenu from '../../QRCodeMenu';
import renderer from 'react-test-renderer';
import {useSyncData} from '../../queryServices/useSyncData';

jest.mock('react-native-encrypted-storage', () => {
  const RNEncryptedStorage = {
    setItem: jest.fn(() => Promise.resolve()),
    getItem: jest.fn(() => Promise.resolve()),
    removeItem: jest.fn(() => Promise.resolve()),
    clear: jest.fn(() => Promise.resolve()),
  };

  return RNEncryptedStorage;
});
jest.mock('react-query', () => ({
  useQueryClient: jest.fn(),
  QueryClientProvider: jest.fn(),
  useMutation: jest.fn(),
}));
jest.mock('../../queryServices/useSyncData', () => ({
  useSyncData: jest.fn(),
}));
interface SyncData {
  syncData?: any;
  isLoading?: boolean;
  isError?: boolean;
  isSuccess?: boolean;
  data?: any;
}

const mockUseSyncData = useSyncData as jest.Mock<SyncData>;
describe('QRCodeMenu', () => {
  const navigation = {navigate: jest.fn()};
  it('renders correctly', () => {
    mockUseSyncData.mockImplementation(() => ({
      isLoading: false,
    }));
    const tree = renderer
      .create(<QRCodeMenu navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
