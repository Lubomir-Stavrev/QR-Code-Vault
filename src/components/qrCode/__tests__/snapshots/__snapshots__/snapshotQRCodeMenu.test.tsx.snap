// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCodeMenu renders correctly 1`] = `
<View
  accessibilityLabel="QRCodeMenuContainer"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#202020",
      "flex": 1,
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Roboto",
            "fontSize": 35,
            "letterSpacing": 1,
          }
        }
      >
        QR Code Vault
      </Text>
      <View>
        <View
          accessibilityLabel="navigateToQRCodeScan"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#01a3ea",
                "borderRadius": 5,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "paddingBottom": 15,
                "paddingTop": 15,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffff",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "700",
                  "letterSpacing": 1,
                }
              }
            >
              Scan QR CODE
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="navigateToUserQRCodes"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#01a3ea",
                "borderRadius": 5,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "paddingBottom": 15,
                "paddingTop": 15,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffff",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "700",
                  "letterSpacing": 1,
                }
              }
            >
              View QR code collection
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="syncData"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#01a3ea",
                "borderRadius": 5,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "paddingBottom": 15,
                "paddingTop": 15,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffff",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "700",
                  "letterSpacing": 1,
                }
              }
            >
              SYNC QR codes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
